stubs/redis/redis/client.pyi

```diff
@@ -355,7 +359,26 @@ class Redis(Generic[_StrType]):
     def set_response_callback(self, command, callback): ...
     def pipeline(self, transaction: bool = ..., shard_hint: Any = ...) -> Pipeline: ...
     def transaction(self, func, *watches, **kwargs): ...
-    def lock(self, name, timeout=..., sleep=..., blocking_timeout=..., lock_class=..., thread_local=...): ...
+    @overload
+    def lock(
+        self,
+        name: _Key,
+        timeout: Optional[float] = ...,
+        sleep: float = ...,
+        blocking_timeout: Optional[float] = ...,
+        lock_class: Optional[Type[Lock]] = ...,
+        thread_local: bool = ...,
+    ) -> Lock: ...
+    @overload
+    def lock(
+        self,
+        name: _Key,
+        timeout: Optional[float] = ...,
+        sleep: float = ...,
+        blocking_timeout: Optional[float] = ...,
+        lock_class: Optional[Type[_LockType]] = ...,
```

Comment:
Not really `Optional` for this overload. Also this overload should not have a default value, but removing it requires two overloads (three total):

```
@overload
def lock([...], lock_class: None = ..., [...]) -> Lock: ...
@overload  # no defaults for arguments before lock_class
def lock(name: _Key, timeout: Optional[float], [...], lock_class: Type[_LockType], [...]) -> _LockType: ...
@overload
def lock(name: _Key, timeout: Optional[float] = ..., [...], *, lock_class: Type[_LockType], [...]) -> _LockType: ...
```

Reply:
Out of curiosity, why should lock_class not have a default?

Reply:
When the default is used, it is `None`, which means that the second overload should not match. It's not super important in this case, as overloads are matched in order anyway, but this could introduce subtle bugs when these definitions are ever changed.